listNetworkVRs.py

```diff
@@ -0,0 +1,225 @@
+        print "ERROR: Invalid filter: %s" % opFilter
+        sys.exit(3)
+
+
+# Parse arguments
+if __name__ == "__main__":
+    handleArguments(sys.argv[1:])
+
+# Init our class
+c = cloudstackops.CloudStackOps(DEBUG, DRYRUN)
+
+if DEBUG == 1:
+    print "# Warning: Debug mode is enabled!"
+
+if DRYRUN == 1:
+    print "# Warning: dry-run mode is enabled, not running any commands!"
+
+# make credentials file known to our class
+c.configProfileName = configProfileName
+
+# Init the CloudStack API
+c.initCloudStackAPI()
+
+if DEBUG == 1:
+    print "API address: " + c.apiurl
+    print "ApiKey: " + c.apikey
+    print "SecretKey: " + c.secretkey
+
+
+def getListNetworks(filter=None, filterNoRR=None):
+    results = []
+    if DEBUG == 1:
+        print '[d] getListNetworks() - networkType = %s, rrType = %s' % (opFilter, opFilterNoRR)
+
+    networkData = c.listNetworks()
+    for network in networkData:
+        rr_type = False
+        if network.service:
+            for netsvc in network.service:
+                if netsvc.capability:
+                    for cap in netsvc.capability:
+                        if cap.name == 'RedundantRouter':
+                            rr_type = True
+
+        routersData = c.getRouterData({'networkid': network.id})
+        routers = []
+        if routersData:
+            for r in routersData:
+                routers = routers + [ r.name ]
+
+        if ( (filter in [None, network.type]) and (filterNoRR in [None, rr_type]) ):
+            results = results + [{ 'id': network.id, 'type': network.type, 'name': network.name, 'domain': network.domain, 'rr_type': rr_type, 'restartrequired': network.restartrequired, 'state': network.state, 'vrs': routers }]
+
+    vpcData = routers = c.listVPCs()
+    for vpc in vpcData:
+        rr_type = False
+        if vpc.redundantvpcrouter:
+            rr_type = vpc.redundantvpcrouter
+
+        routersData = c.getRouterData({'vpcid': vpc.id})
+        routers = []
+        if routersData:
+            for r in routersData:
+                routers = routers + [ r.name ]
+
+        if ( (filter in [None, 'VPC']) and (filterNoRR in [None, rr_type]) ):
+             results = results + [{ 'id': vpc.id, 'type': 'VPC', 'name': vpc.name, 'domain': vpc.domain, 'rr_type': rr_type, 'restartrequired': vpc.restartrequired, 'state': vpc.state, 'vrs': routers }]
+
+    return results
+
+
+def cmdListNetworks():
+    networkData = getListNetworks(opFilter, opFilterNoRR)
+    counter = 0
+
+#    import pprint
+#    pp = pprint.PrettyPrinter(indent=4)
+#    pp.pprint(networkData)
+    
+    # Empty line
+    print
+    t = PrettyTable(["#", "Network", "Type", "ID", "Domain", "State", "Redundant?", "RestartReq?", "VRs"])
+    #t.align["VM"] = "l"
+
+    for n in networkData:
+        counter = counter + 1
+
+        t.add_row([counter, n['name'], n['type'], n['id'], n['domain'], n['state'], n['rr_type'], n['restartrequired'], n['vrs']])
```

Comment:
Would be nice if you could print the VRs a bit prettier ;-)

![screen shot 2016-03-13 at 18 23 01 pm](https://cloud.githubusercontent.com/assets/1630096/13730179/a6bbb4b6-e948-11e5-9104-f56f15634264.png)